salt/kubelet/kubelet.jinja

```diff
@@ -32,5 +44,6 @@ KUBELET_ARGS="\
     --cloud-provider=aws \
 {% endif -%}
     {{ pillar['components']['kubelet']['args'] }} \
+    --node-labels={{ node_labels|join(',') }} \
```

Comment:
I think the way it's setup, there was always be values.. However, wrapping it in a `{% if node_labels %} .. {% endif %}` is harmless and prevents it breaking if `--node-labels=` would otherwise cause an issue..

Also, this will only set the labels once, on first startup (well, during node registration). Any post install kubelet start changes won't be applied. And, if the labels change, we'll unnecessarily restart the kubelet daemon. We hit this in the HCP ansible port, and solved it using a mix of two things:

1. Render the labels args into a separate file, e.g. `/etc/kuebrnetes/kubelet-initial`, do not watch this file from any services. Include this file into this one, and append the included `$KUBELET_ARGS` to `KUBELET_ARGS`. This means labels get set on the initial startup, and no unnecessary restarts will happen.

2. To handle future label updates, we had to do something like this for each label: `kubectl --kubeconfig /srv/kubernetes/kubeconfig label --overwrite node {{ inventory_hostname }} {{ item.key }}={{ item.value }}` (we had a TODO to make this a ansible module rather than a shell out, maybe salt has a module ready to use?)